File: facebook/src/main/java/com/facebook/FacebookDialogException.java
/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * You are hereby granted a non-exclusive, worldwide, royalty-free license to use,
 * copy, modify, and distribute this software in source code or binary form for use
 * in connection with the web services and APIs provided by Facebook.
 *
 * As with any software that integrates with the Facebook platform, your use of
 * this software is subject to the Facebook Developer Principles and Policies
 * [http://developers.facebook.com/policy/]. This copyright notice shall be
 * included in all copies or substantial portions of the software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
 * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
 * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
 * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

package com.facebook;

/**
 * Represents an error condition relating to displaying a Facebook Web dialog.
 */
public class FacebookDialogException extends FacebookException {
    static final long serialVersionUID = 1;
    private int errorCode;
    private String failingUrl;

    /**
     * Constructs a new FacebookException.
     */
    public FacebookDialogException(String message, int errorCode, String failingUrl) {
        super(message);
        this.errorCode = errorCode;
        this.failingUrl = failingUrl;
    }

    /**
     * Gets the error code received by the WebView. See:
     * http://developer.android.com/reference/android/webkit/WebViewClient.html
     *
     * @return the error code
     */
    public int getErrorCode() {
        return errorCode;
    }

    /**
     * Gets the URL that the dialog was trying to load.
     * @return the URL
     */
    public String getFailingUrl() {
        return failingUrl;
    }

    @Override
    public final String toString() {
        return new StringBuilder()
                .append("{FacebookDialogException: ")
                .append("errorCode: ")
                .append(getErrorCode())
                .append(", message: ")
                .append(getMessage())
                .append(", url: ")
                .append(getFailingUrl())
                .append("}")
                .toString();
    }
}

File: facebook-common/src/main/java/com/facebook/FacebookDialogException.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook

/** Represents an error condition relating to displaying a Facebook Web dialog. */
class FacebookDialogException
/** Constructs a new FacebookException. */
(
    message: String?,
    /**
     * The error code received by the WebView. See:
     * http://developer.android.com/reference/android/webkit/WebViewClient.html
     */
    val errorCode: Int,
    /** The URL that the dialog was trying to load. */
    val failingUrl: String?
) : FacebookException(message) {
  override fun toString(): String {
    return StringBuilder()
        .append("{FacebookDialogException: ")
        .append("errorCode: ")
        .append(errorCode)
        .append(", message: ")
        .append(message)
        .append(", url: ")
        .append(failingUrl)
        .append("}")
        .toString()
  }

  companion object {
    const val serialVersionUID: Long = 1
  }
}
